staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -799,8 +869,8 @@ func DefaultBuildHandlerChain(apiHandler http.Handler, c *Config) http.Handler {
 	handler = genericapifilters.WithWarningRecorder(handler)
 	handler = genericapifilters.WithCacheControl(handler)
 	handler = genericfilters.WithHSTS(handler, c.HSTSDirectives)
-	if c.ShutdownSendRetryAfter {
-		handler = genericfilters.WithRetryAfter(handler, c.lifecycleSignals.AfterShutdownDelayDuration.Signaled())
+	if shouldRespondWithRetryAfterFn, ok := c.shouldAddWithRetryAfterFilter(); ok {
```

Comment:
this conditional seems unnecessary complicated. Why not apply the filter always and make the predicate clever enough?